src/PrestaShopBundle/Resources/config/services/adapter/product.yml

```diff
@@ -65,3 +65,12 @@ services:
         tags:
             - name: tactician.handler
               command: PrestaShop\PrestaShop\Core\Domain\Product\Query\GetProductIsEnabled
+
+    prestashop.adapter.product.command_handler.add_product_handler:
+      class: PrestaShop\PrestaShop\Adapter\Product\CommandHandler\AddProductHandler
+      arguments:
+        - "@=service('prestashop.adapter.legacy.configuration').getInt('PS_LANG_DEFAULT')"
+        - '@=service("prestashop.adapter.legacy.context").getContext().shop.id_category'
```

Comment:
What if there is no id_category in the shop context? :thinking:

Reply:
The `id_category` is required in the `Shop` object/table so unless the DB is corrupted in a way it should be hee, I don't think we should overly complicate the service definition for a use case that implies messing with the DB

Although it would be better to have a dedicated service to get these multi shop values, I guess it could be managed properly when the multistore is refactored in 178? But not in this PR